mempool/mempool.go

```diff
@@ -44,6 +44,10 @@ type Mempool interface {
 	// (~ all available transactions).
 	ReapMaxTxs(max int) types.Txs
 
+	// FilterTx returns the types.Tx with the given hash if found in the mempool,
+	// otherwise returns nil.
+	FilterTx(hash []byte) types.Tx
```

Comment:
This is just a personal preference but I would change the name of the function to something more specific like `GetTxByHash`.